drawee/src/main/java/com/facebook/drawee/drawable/RoundedCornersDrawable.java

```diff
@@ -145,16 +154,50 @@ protected void onBoundsChange(Rect bounds) {
 
   private void updatePath() {
     mPath.reset();
+    mBorderPath.reset();
     mTempRectangle.set(getBounds());
+    switch (mType){
+      case CLIPPING:
+        mTempRectangle.inset(mPadding, mPadding);
+        if (mIsCircle) {
+          mPath.addCircle(
+                  mTempRectangle.centerX(),
+                  mTempRectangle.centerY(),
+                  Math.min(mTempRectangle.width(), mTempRectangle.height()) / 2,
+                  Path.Direction.CW);
+        } else {
+          mPath.addRoundRect(mTempRectangle, mRadii, Path.Direction.CW);
+        }
+        mTempRectangle.inset(-mPadding, -mPadding);
+        break;
+      case OVERLAY_COLOR:
+        mTempRectangle.inset(mPadding, mPadding);
+        if (mIsCircle) {
+          mPath.addCircle(
+                  mTempRectangle.centerX(),
+                  mTempRectangle.centerY(),
+                  Math.min(mTempRectangle.width(), mTempRectangle.height()) / 2,
+                  Path.Direction.CW);
+        } else {
+          mPath.addRoundRect(mTempRectangle, mRadii, Path.Direction.CW);
+        }
+        mTempRectangle.inset(-mPadding, -mPadding);
+        break;
+    }
+
```

Comment:
This whole switch case seems unnecessary as the logic is same in all cases. Can you just update the block below instead?